jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java

```diff
@@ -366,8 +366,11 @@ public boolean handle()
                 {
                     case TERMINATED:
                         onCompleted();
-                        _request.getHttpInput().failed(new IOException("terminated"));
-                        getEndPoint().cancelFillInterested();
+                        if (_request.getHttpInput().isError())
+                        {
+                            _request.getHttpInput().failed(new IOException("terminated"));
```

Comment:
If we are already in error, then why do we need to fail again?   Any caller will not block in read and get an exception either way.

Reply:
If we blindly fail the input upon termination, any content already parsed but not yet consumed by the servlet will be discarded and make the servlet throw an exception while there was no error.

This extra hack seems to fix a good chunk of the tests that failed during the previous build, it just adds to the cleanup that will be needed to get this PR to a reasonable state.

Reply:
I don't doubt we need to be smart here and not break chained requests.  I was thinking more that if we are at EOF then we do nothing.
What I don't understand is how replacing and error with another error is helping? Is it just the cancel on the next line?
I think we need to understand why this works.